test/ioextras/abstract_output_stream_test.rb

```diff
@@ -50,19 +48,18 @@ def test_print
     @output_stream.print(' You ok ', 'out ', 'there?')
     assert_equal('hello world. You ok out there?', @output_stream.buffer)
 
-    $\ = "\n"
+    $OUTPUT_RECORD_SEPARATOR = "\n"
     @output_stream.print
     assert_equal("hello world. You ok out there?\n", @output_stream.buffer)
 
     @output_stream.print('I sure hope so!')
     assert_equal("hello world. You ok out there?\nI sure hope so!\n", @output_stream.buffer)
 
-    $, = 'X'
     @output_stream.buffer = ''
     @output_stream.print('monkey', 'duck', 'zebra')
-    assert_equal("monkeyXduckXzebra\n", @output_stream.buffer)
+    assert_equal("monkeyduckzebra\n", @output_stream.buffer)
```

Comment:
I guess this test doesn't make any sense now? It's there purely to test that `AbstractOutputStream` respects `$,`. We have a test above (`@output_stream.print(' You ok ', 'out ', 'there?')`) which tests `#print` with multiple strings.

Should we remove this?

Reply:
I think this PR can ship without this further change. My intention was simply to remove the deprivation which this PR accomplishes. If we want to remove this test, we can do so in another PR. 

Personally, I don't have an opinion on whether or not the test should be removed.

Reply:
OK. I had to resolve a conflict on those lines anyway, so I got rid of them then.